avocado/core/messages.py

```diff
@@ -147,10 +147,33 @@ class BaseRunningMessageHandler(BaseMessageHandler):
     """Base interface for resolving running messages."""
 
     @staticmethod
-    def _save_message_to_file(filename, buff, task, mode='a'):
+    def _save_message_to_file(filename, buff, task):
+        """
+        Method for saving messages into the file
+
+        It can save str or byte types of messages. When the str message doesn't
+        end with a new line the new line will be added. Every message is saved
+        in the append mode.
+
+        :param filename: name of the file
+        :type filename: str
+        :param buff: message to be saved
+        :type buff: str or bytes
+        :param task: message related task.
+        :type: :class:`avocado.core.nrunner.Task`
+        """
+
+        def _save_to_file(mode):
+            with open(file, mode) as fp:
```

Comment:
I understand that `file` will be evaluated later, but I find it extremely confusing to refer to a variable in a different scope that has not been defined yet (in the code order).

Maybe it's just me, but I'd like to hear from others (@willianrampazzo @beraldoleal).

Reply:
Indeed, also `_save_to_file()` seems to me that is waiting for a file to save. I would move the filename as the first argument here.

Reply:
It looks weird for me too, but as this is a function inside a method, it has access to its context. Shame on me. I also used this construction before :P